Marlin/src/lcd/marlinui.cpp

```diff
@@ -1035,9 +1035,9 @@ void MarlinUI::init() {
         uint8_t abs_diff = ABS(encoderDiff);
 
         #if ENCODER_PULSES_PER_STEP > 1
-          static int8_t lastEncoderDiff;
+          TERN_(HAS_TOUCH_SLEEP, static int8_t lastEncoderDiff);
           TERN_(HAS_TOUCH_SLEEP, if (lastEncoderDiff != encoderDiff) wakeup_screen());
-          lastEncoderDiff = encoderDiff;
+          TERN_(HAS_TOUCH_SLEEP, lastEncoderDiff = encoderDiff);
         #endif
```

Comment:
how about changing it to this?

```diff
         #if ENCODER_PULSES_PER_STEP > 1
-          TERN_(HAS_TOUCH_SLEEP, static int8_t lastEncoderDiff);
-          TERN_(HAS_TOUCH_SLEEP, if (lastEncoderDiff != encoderDiff) wakeup_screen());
-          TERN_(HAS_TOUCH_SLEEP, lastEncoderDiff = encoderDiff);
         #endif

         #if ENCODER_PULSES_PER_STEP > 1
+          #if HAS_TOUCH_SLEEP
+            static int8_t lastEncoderDiff);
+            if (lastEncoderDiff != encoderDiff) wakeup_screen());
+            lastEncoderDiff = encoderDiff);
+          #endif
         #endif

Reply:
Sounds good.  TEh change is made

Reply:
I realize I have made a small mistake above, at the end of the three middle lines there is an extra `)` which should be deleted